src/idl_parser.cpp

```diff
@@ -1255,7 +1255,10 @@ CheckedError Parser::ParseTable(const StructDef &struct_def, std::string *value,
                 CTYPE val, valdef; \
                 ECHECK(atot(field_value.constant.c_str(), *this, &val)); \
                 ECHECK(atot(field->value.constant.c_str(), *this, &valdef)); \
-                builder_.AddElement(field_value.offset, val, valdef); \
+                if(field->key) \
```

Comment:
I don't understand how this fixes it.. this may fix it for JSON data but the problem is still there when using binary data created elsewhere?

I assumed that the fix had to be in the actual sorting code.

Reply:
This problem arises in `Parser::ParseVector` only.
Binary data never use `SimpleQsort`.
`CreateVectorOfSortedTables` and `CreateVectorOfSortedStructs` use `std::sort` with comparators.
https://github.com/google/flatbuffers/blob/0800976533ac2b54790635a5eef8b5d9dbc5e8eb/include/flatbuffers/flatbuffers.h#L1982-L-1987.

But you are right, the `Parser::ParseTable` isn't a good place for this fix.

Reply:
I think `key` fields should be implicitly declared as `required` fields (both string and scalars).
At least it should be `required` if a struct or a table is used inside `Vector` container.
We can set it as `required` in FBS-2.0.

Reply:
I moved `key` checking into `Parser::ParseVector`.
CI failure doesn't connect with these changes.

Reply:
Yes, `key` implying `required` makes a lot of sense for strings. For scalars, it shouldn't really matter, though if it simplifies our key lookup, let's do it.